src/main/java/fi/dy/masa/minihud/renderer/OverlayRendererVillagerOffers.java

```diff
@@ -0,0 +1,118 @@
+package fi.dy.masa.minihud.renderer;
+
+import fi.dy.masa.malilib.render.RenderUtils;
+import fi.dy.masa.malilib.util.WorldUtils;
+import fi.dy.masa.minihud.config.Configs;
+import fi.dy.masa.minihud.config.RendererToggle;
+import fi.dy.masa.minihud.mixin.IMixinMerchantEntity;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.component.DataComponentTypes;
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.passive.VillagerEntity;
+import net.minecraft.item.Items;
+import net.minecraft.registry.entry.RegistryEntry;
+import net.minecraft.registry.tag.EnchantmentTags;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.village.TradeOffer;
+import net.minecraft.village.TradeOfferList;
+import net.minecraft.village.VillagerProfession;
+import net.minecraft.world.World;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OverlayRendererVillagerOffers extends OverlayRendererBase
+{
+    public static final OverlayRendererVillagerOffers INSTANCE = new OverlayRendererVillagerOffers();
+    @Override
+    public String getName()
+    {
+        return "Villager Offers Overlay";
+    }
+
+    @Override
+    public boolean shouldRender(MinecraftClient mc)
+    {
+        return RendererToggle.OVERLAY_VILLAGER_OFFERS.getBooleanValue();
+    }
+
+    @Override
+    public boolean needsUpdate(Entity entity, MinecraftClient mc)
+    {
+        return true;
+    }
```

Comment:
Can we check to see that this doesn't cause the renderer to constantly be refreshed?
The other Renderer's has a 'setNeedsUpdate()' function.

Reply:
I am not sure, I tried to update this renderer every 10 ticks and it seems only render overlay per 10 ticks. I am using CPU to render this overlay, thus, render = update (I guess "update" refers to update vbo stuff). This renderer uses Minecraft's text renderer so it can only rendered by CPU I guess.